ldap/jwt.go

```diff
@@ -0,0 +1,101 @@
+package ldap
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	jwt "github.com/dgrijalva/jwt-go"
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/log"
+)
+
+// GenerateToken generates JWT
+func GenerateToken(user User, authConfig config.AuthConfig) (Token, error) {
+	signingKey := config.Get().LoginToken.SigningKey
+
+	// Create the token
+	token := jwt.New(jwt.SigningMethodHS256)
+	expirationTime := time.Now().Add(time.Second * time.Duration(config.Get().LoginToken.ExpirationSeconds))
+	// Create a map to store our claims
+	claims := token.Claims.(jwt.MapClaims)
+
+	// Set token claims
+	claims["username"] = user.Username
+	claims["uid"] = user.UID
+	claims["groups"] = user.Groups
+	claims["exp"] = expirationTime.Unix()
+	claims["iat"] = time.Now().Unix()
+
+	signedToken, err := token.SignedString([]byte(signingKey))
+	if err != nil {
+		log.Errorf("Cannot sign token  : %s", err)
+		return Token{}, err
+	}
+	return Token{
+		JWT:    signedToken,
+		Expiry: expirationTime,
+	}, nil
+}
+
+// ValidateToken validates JWT token provided by user and fills out the UserInfo structure from the data within
+func ValidateToken(token string) (UserInfo, error) {
+	return validate(token)
+}
+
+// validate does much of the work of ValidateToken
+func validate(bearerToken string) (UserInfo, error) {
+
+	signingKey := config.Get().LoginToken.SigningKey
+
+	var auth bool
```

Comment:
This may be more clear as:  `auth := false`

Reply:
done